controller/lib/include/descriptor_base.h

```diff
@@ -52,6 +52,19 @@ class descriptor_base
     /// \return The index of the descriptor.
     ///
     AVDECC_CONTROLLER_LIB32_API virtual uint16_t STDCALL descriptor_index() const = 0;
+    
+    ///
+    /// Check the descriptor's permission status.
+    ///
+    /// \param flag The LOCK_ENTITY or ACQUIRE_ENTITY flag to retrieve.
+    ///        Valid flags are 0 (LOCK, ACQUIRE), 0x1 (UNLOCK, PERSISTENT), and 0x80000000 (RELEASE).
+    /// \see avdecc_lib::permission_flags
+    /// \param cmd_type Valid types are 0 (AEM_CMD_ACQUIRE_ENTITY) and 1 (AEM_CMD_LOCK_ENTITY)
+    ///
+    /// \return True if the current permission status matches the flag.
+    ///         False if the current permission status differs.
+    ///
+    AVDECC_CONTROLLER_LIB32_API virtual bool STDCALL get_permisson(int flag, int cmd_type) = 0;
```

Comment:
Why do we have to pass in the cmd type? Can we have that as part of the name of the flag we are going to extract?